### PR TITLE
[MEMKEYDB]Feat add building memkind

### PR DESCRIPTION
**Pull request by michalbiesek**
This change is [Reviewable](https://reviewable.io/reviews/pmem/redis/168)

### CODE REVIEW

**Review by kasiawasiuta — COMMENTED** [review]
:lgtm: 

> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/pmem/redis/168#-:-M4ny0E256uMHrXkjR5T:bg7igkv)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jschmieg)

File: src/Makefile [hunk 1]
```diff
@@ -70,6 +70,10 @@ ifeq ($(USE_JEMALLOC),no)
 	MALLOC=libc
 endif
 
+ifeq ($(USE_MEMKIND),yes)
```
Comment (guoanwu) [inline src/Makefile:66]: we can let the memkind as the default malloc and update the read me.

Comment (michalbiesek) [reply src/Makefile:66]: Done. Memkind is used as default allocator

**Comment by jschmieg** [comment]
Is it possible to specify the version of memkind in INFO MEMORY response, like it is done with jemalloc? Is this exposed from building process?
mem_allocator:jemalloc-5.1.0
mem_allocator:memkind

**Review by michalbiesek — COMMENTED** [review]
This would require changes in libmemkind. The information is exposed from libmemkind and must to be know during redis compilation process. Redis doesn't use the pkgconfig but I willtake a look for what should be change in libmemkind to achieve this. Just let me know it if it is required here - if yes then if should be added now or could be done in separate PR.

> *[Reviewable](https://reviewable.io/reviews/pmem/redis/168#-:-M4yCYIX24yJN5u0p7rG:b-qk434h)* status: 5 of 7 files reviewed, 1 unresolved discussion (waiting on @guoanwu, @jschmieg, and @kasiawasiuta)

**Review by jschmieg — APPROVED** [review]

**Comment by michalbiesek** [comment]
Merged in memkeydb.

**Pull request closed without merging**